File: leetcode/majority-element.py
// https://leetcode.com/problems/majority-element

class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        n = len(nums)
        cnt = {}
        for num in nums:
            if num not in cnt:
                cnt[num] = 0
            cnt[num] += 1
        
        for num, appearTimes in cnt.items():
            if appearTimes > n/2:
                return num
        return -1

File: leetcode/majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n = nums.size();
        unordered_map<int, int> cnt;
        for (int x : nums)
            if (++cnt[x] > n/2)
                return x;
        return 0;
    }
};